nosnoc/problem.py

```diff
@@ -456,12 +467,26 @@ def create_complementarity(self, x: List[ca.SX], y: ca.SX, sigma: ca.SX, tau: ca
 
         return
 
-    def create_complementarity_constraints(self, sigma_p: ca.SX, tau: ca.SX) -> None:
+    def create_complementarity_constraints(self, sigma_p: ca.SX, tau: ca.SX, Uk: ca.SX) -> None:
         opts = self.opts
+        # handle path complementarities TODO maintain sparsity?
+        # TODO: This needs also to provide a lower bound on a,b in a smart way
+        #       however this needs to be done carefully to maintain LICQ
+        X_fe = self.X_fe()
+        for j in range(opts.n_s):
+            z = self.w[self.ind_z[j]]
+            stage_comps = self.ocp.g_stage_comp_fun(X_fe[j], Uk, z, self.p, self.model.v_global)  # TODO maybe should include stage z
```

Comment:
`# TODO maybe should include stage z`
Please just go for one and remove the todo.